records: allow for image files without thumbnails

GraemeWatt opened the pull request:
If there is no matching thumbnail available for an image file, just use the image file itself as the preview location.  I've tested on my local machine that the changes work correctly.  Closes #459.

GraemeWatt committed:
records: allow for image files without thumbnails

* Fixes #459.

alisonrclarke reviewed (approved):
Looks fine; just wonder if the code could be easier to read if we modified the original loop to use the actual file as the preview location and keep track of thumbs, then update with thumbnails in a second loop?

GraemeWatt committed:
records: refactor default preview_location code

* Also now allow for a thumbnail file without a matching image file.

GraemeWatt commented:
> Looks fine; just wonder if the code could be easier to read if we modified the original loop to use the actual file as the preview location and keep track of thumbs, then update with thumbnails in a second loop?

Good idea, I've refactored the code slightly.  I've also allowed for the special case of a thumbnail file without a matching image file.  I couldn't work out an elegant way to do the latter, so feel free to make another commit if you think the code is still too messy.

GraemeWatt committed:
Merge branch 'master' into fix-image-thumbnail
